src/cli/trn_cli_common.c

```diff
@@ -617,3 +617,68 @@ int trn_cli_parse_ebpf_prog_stage(const cJSON *jsonobj,
 	}
 	return 0;
 }
+
+int trn_cli_parse_network_policy_cidr(const cJSON *jsonobj,
+				      struct rpc_trn_vsip_cidr_t *cidrval)
+{
+	cJSON *tunnel_id = cJSON_GetObjectItem(jsonobj, "tunnel_id");
+	cJSON *local_ip = cJSON_GetObjectItem(jsonobj, "local_ip");
+	cJSON *cidr_prefixlen = cJSON_GetObjectItem(jsonobj, "cidr_prefixlen");
+	cJSON *cidr_ip = cJSON_GetObjectItem(jsonobj, "cidr_ip");
+	cJSON *cidr_type = cJSON_GetObjectItem(jsonobj, "cidr_type");
+	cJSON *bit_val = cJSON_GetObjectItem(jsonobj, "bit_value");
+
+	if (tunnel_id == NULL) {
+		cidrval->tunid = 0;
+	} else if (cJSON_IsString(tunnel_id)) {
+		cidrval->tunid = atoi(tunnel_id->valuestring);
+	} else {
+		print_err("Error: Network policy tunnel_id is non-string.\n");
+		return -EINVAL;
+	}
+
+	if (local_ip != NULL && cJSON_IsString(local_ip)) {
+		cidrval->local_ip = parse_ip_address(local_ip);
+	} else {
+		print_err("Error: Network policy local IP is missing or non-string\n");
+		return -EINVAL;
+	}
+
+	if (cidr_ip != NULL && cJSON_IsString(cidr_ip)) {
+		cidrval->cidr_ip = parse_ip_address(cidr_ip);
+	} else {
+		print_err("Error: Network policy CIDR IP is missing or non-string\n");
+		return -EINVAL;
+	}
+
+	// adding 96 = 0x01100000 offsets the missing bit
```

Comment:
1. please explain the magic number 96 a bit.
2. what about putting the addition to daemon, instead of here at client?